epoxy-viewbinder/src/main/java/com/airbnb/epoxy/EpoxyViewBinder.kt

```diff
@@ -0,0 +1,480 @@
+        }
+    }
+
+    private fun EpoxyModel<*>.hasSameViewType(model: EpoxyModel<*>): Boolean =
+        ViewTypeManager.getViewType(this) == ViewTypeManager.getViewType(model)
+}
+
+/**
+ * Shortcut for creating a [LifecycleAwareEpoxyViewBinder] in a lazy way. 
+ * See [LifecycleAwareEpoxyViewBinder] for pass-through parameter documentation.
+ *
+ * @param initializer a lambda that is run directly after instantiation of the
+ * [LifecycleAwareEpoxyViewBinder].
+ */
+fun ComponentActivity.epoxyView(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit,
+    modelProvider: EpoxyController.() -> Unit
+) = lazy {
+    return@lazy epoxyViewInternal(viewId, useVisibilityTracking, initializer, modelProvider)
+}
+
+/**
+ * Shortcut for creating a [LifecycleAwareEpoxyViewBinder] in a lazy way.
+ * See [LifecycleAwareEpoxyViewBinder] for pass-through parameter documentation.
+ *
+ * @param initializer a lambda that is run directly after instantiation of the
+ * [LifecycleAwareEpoxyViewBinder].
+ */
+fun Fragment.epoxyView(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit,
+    modelProvider: EpoxyController.() -> Unit
+) = lazy {
+    return@lazy epoxyViewInternal(viewId, useVisibilityTracking, initializer, modelProvider)
+}
+
+/**
+ * Shortcut for creating a [LifecycleAwareEpoxyViewBinder] in a lazy way.
+ * See [LifecycleAwareEpoxyViewBinder] for pass-through parameter documentation.
+ *
+ * @param initializer a lambda that is run directly after instantiation of the
+ * [LifecycleAwareEpoxyViewBinder].
+ */
+fun ViewGroup.epoxyView(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit,
+    modelProvider: EpoxyController.() -> Unit
+) = lazy {
+    return@lazy epoxyViewInternal(viewId, useVisibilityTracking, initializer, modelProvider)
+}
+
+/**
+ * Shortcut for creating a [LifecycleAwareEpoxyViewBinder] in a lazy way.
+ * See [LifecycleAwareEpoxyViewBinder] for pass-through parameter documentation.
+ *
+ * @param initializer a lambda that is run directly after instantiation of the
+ * [LifecycleAwareEpoxyViewBinder].
+ *
+ * @return a view binder or null if a view with the [viewId] could not be found.
+ */
+fun ComponentActivity.optionalEpoxyView(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    fallbackToNameLookup: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit = { },
+    modelProvider: EpoxyController.() -> Unit
+) = lazy {
+    val view = findViewById<View>(android.R.id.content)
+    // View id is not present, we just return null in that case.
+    if (view.maybeFindViewByIdName<View>(viewId, fallbackToNameLookup) == null) return@lazy null
+
+    return@lazy epoxyViewInternal(viewId, useVisibilityTracking, initializer, modelProvider)
```

Comment:
Good call out. These are performing their own `maybeFindViewByIdName` calls. Those should probably just use the one that exists in `LifecycleAwareEpoxyViewBinder`.

Reply:
Actually that would then crash. I think on how to clean this up

Reply:
Kept the `maybeFindViewByIdName` calls here because of implication with view nullability in `LifecycleAwareEpoxyViewBinder`. I forwarded the `fallbackToNameLookup` field though for parity